crates/shielder-scheduler-server/src/storage/dynamo_db.rs

```diff
@@ -0,0 +1,475 @@
+
+        let attr_def_id = AttributeDefinition::builder()
+            .attribute_name("id")
+            .attribute_type(ScalarAttributeType::S)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("AttrDef build error: {e}")))?;
+
+        let attr_def_status = AttributeDefinition::builder()
+            .attribute_name("status")
+            .attribute_type(ScalarAttributeType::S)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("AttrDef build error: {e}")))?;
+
+        let attr_def_relay_after = AttributeDefinition::builder()
+            .attribute_name("relay_after")
+            .attribute_type(ScalarAttributeType::N)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("AttrDef build error: {e}")))?;
+
+        let key_schema = KeySchemaElement::builder()
+            .attribute_name("id")
+            .key_type(KeyType::Hash)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("KeySchema build error: {e}")))?;
+
+        // GSI key schema
+        let gsi_hash_key = KeySchemaElement::builder()
+            .attribute_name("status")
+            .key_type(KeyType::Hash)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("GSI KeySchema build error: {e}")))?;
+
+        let gsi_range_key = KeySchemaElement::builder()
+            .attribute_name("relay_after")
+            .key_type(KeyType::Range)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("GSI KeySchema build error: {e}")))?;
+
+        let throughput = ProvisionedThroughput::builder()
+            .read_capacity_units(5)
+            .write_capacity_units(5)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("Throughput build error: {e}")))?;
+
+        let gsi_throughput = ProvisionedThroughput::builder()
+            .read_capacity_units(5)
+            .write_capacity_units(5)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("GSI Throughput build error: {e}")))?;
+
+        let projection = Projection::builder()
+            .projection_type(ProjectionType::All)
+            .build();
+
+        let gsi = GlobalSecondaryIndex::builder()
+            .index_name(STATUS_RELAY_INDEX)
+            .key_schema(gsi_hash_key)
+            .key_schema(gsi_range_key)
+            .projection(projection)
+            .provisioned_throughput(gsi_throughput)
+            .build()
+            .map_err(|e| StorageError::Internal(format!("GSI build error: {e}")))?;
+
+        if let Err(e) = self
+            .client
+            .create_table()
+            .table_name(TABLE_NAME)
+            .attribute_definitions(attr_def_id)
+            .attribute_definitions(attr_def_status)
+            .attribute_definitions(attr_def_relay_after)
+            .key_schema(key_schema)
+            .global_secondary_indexes(gsi)
+            .provisioned_throughput(throughput)
```

Comment:
Let's not use provisioned mode, but on-demand (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/on-demand-capacity-mode.html), and set up https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/on-demand-capacity-mode-max-throughput.html . This way we pay for what we use, and we are protected from high cost attacks etc